### ISSUE TITLE
Organisations in the bibliography not in alphabetical order

### ISSUE THREAD

**Issue by 772** [JabRef/jabref#11725]
### JabRef version

5.15 (latest release)

### Operating system

GNU / Linux

### Details on version and operating system

Fedora 40

### Checked with the latest development build (copy version output from About dialog)

- [x] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Set IsSortByPosition="false"
3. Use "{Linux Foundation}" as an author.

The bibliography is broken now and not in alphabetical order. Using {}-brackets for organisations was suggested here: https://discourse.jabref.org/t/how-to-enter-corporation-name-in-author-field/2080

### Appendix

![grafik](https://github.com/user-attachments/assets/e9461108-b2ba-4c2e-a871-4a7adffc51ab)

**Comment by Siedlerchr** [comment]
@subhramit  This is jstyle but as you are deeper in the topic now can you take a look please?

**Comment by subhramit** [comment]
> @subhramit  This is jstyle but as you are deeper in the topic now can you take a look please?

Yeah sure

**Comment by subhramit** [comment]
As it looks at first sight, the order is overridden because of the curly braces {}. JabRef already stores them in BibTeX format, which puts braces around them, so when explicitly added, it becomes {{organization name}}. Instead of "o", the first character to participate in the comparator is then "{".

Solution - we will have to ignore extra braces.

**Comment by Siedlerchr** [comment]
We have a RemoveEnclosingBracesFormatter that can be used

**Comment by subhramit** [comment]
![image](https://github.com/user-attachments/assets/0723c9d7-1de6-4187-815a-e53947d2ed86)
Weirdly enough, I can reproduce with {The Geany contributors} but not {Linux Foundation}. It is as if the issue is occurring in some special case.

**Comment by 772** [comment]
Luckily, I have dozens of foundations and companies in my LibreOffice document. :) This is the pattern:

If {}-braces are used, JabRef takes the **last word** within the {}-brackets and sorts it like the ASCII-table. This is why "c" from contributors is above all.

{Linux Foundation} will be sorted in a wrong order, too, if you add two authors with "F" in their last name and "Foundation" is between them.

**Comment by koppor** [comment]
Two thoughts:

- `org.jabref.model.entry.BibEntry#getFieldLatexFree` should be used as first try (if it is OK that "The" comes first). However, that ignores authors completely
- `org.jabref.model.entry.AuthorList#getAsLastFirstFirstLastNamesWithAnd` or similar method should be used. One needs to look at other methods for sorting...

**Comment by ryan-carpenter** [comment]
> Solution - we will have to ignore extra braces, if explicitly present. I'll take a look at this.

Remember that the braces may be needed to protect the formatting of organisation names, even if the name is only one word.

**Comment by ryan-carpenter** [comment]
> The best workaround I found so far is to replace each " " with a rare symbol like "℗". Before finishing your document, you can simply replace all "℗" with " " in LibreOffice.

Have you tried using no-break spaces in the organisation names?
`The~Geany~contributors` should make the whole name behave as one word (and retain "The" in citation key).

Also, note that (without no-break spaces) capitalisation matters, at least in citation keys. `{Linux Foundation} and `{The Geany Contributors}` produce "LinuxFoundation" and "GeanyContributors", respectively in the citation key, but `{Linux foundation}` and `{The Geany foundation}` produce "Linux" and "Geany". I am not sure if this affects the bibliography.

**Comment by 772** [comment]
> Have you tried using no-break spaces in the organisation names?
> `The~Geany~contributors` should make the whole name behave as one word (and retain "The" in citation key).

Yes, it has no effect. When I use ```The~Geany~contributors``` or ```{The~Geany~contributors}```, the sorting is always wrong. It looks like the "~" is only affecting the display in the JabRef GUI but not text in the LibreOffice document.

**Comment by koppor** [comment]
This issue could be related: https://github.com/JabRef/jabref/issues/7805

**Comment by sryjksu** [comment]
@koppor I'm still working on this issue for my final project. Could you please assign this issue to me again? Thank you.

**Comment by sryjksu** [comment]
@subhramit I saw the instruction said that I need to write unit tests for any modification I make. Since this bug is about UI, how do you want the unit test to be like? Also, do you care about coverage of the tests? If so, what types of coverage?

**Comment by subhramit** [comment]
> @subhramit I saw the instruction said that I need to write unit tests for any modification I make. Since this bug is about UI, how do you want the unit test to be like? Also, do you care about coverage of the tests? If so, what types of coverage?

No, unit tests aren't needed for UI changes. Only if there is some change in logic made in the viewmodel, write tests.
